tsdb/chunkenc/histogram_meta.go

```diff
@@ -230,15 +241,15 @@ type Interjection struct {
 // spans themselves, thanks to the iterators we get to work with the more useful
 // bucket indices (which of course directly correspond to the buckets we have to
 // adjust).
-func compareSpans(a, b []histogram.Span) ([]Interjection, bool) {
+func compareSpans(a, b []histogram.Span, includeBackward bool) (forward, backward []Interjection, ok bool) {
```

Comment:
Since more than half of the code is for the `includeBackward` case, maybe it's easier to create two functions along the lines `forwardCompareSpans` and `bidirectionalCompareSpans` or something…

Reply:
Done. With finding merged spans in compareSpans, it makes more sense to split it up.